src/lib/Storage/DoctrineGateway.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzSupportTools\Storage;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ParameterType;
+use eZ\Publish\SPI\Persistence\Content\ContentInfo;
+use eZ\Publish\SPI\Persistence\Content\Type;
+
+class DoctrineGateway implements Gateway
+{
+    private const CONTENTOBJECT_TABLE = 'ezcontentobject';
+    private const USER_TABLE = 'ezuser';
+    private const CONTENT_TYPE_TABLE = 'ezcontentclass';
+    private const CONTENTOBJECT_VERSION_TABLE = 'ezcontentobject_version';
+
+    /** @var \Doctrine\DBAL\Connection */
+    private $connection;
+
+    /** @var \Doctrine\DBAL\Platforms\AbstractPlatform */
+    private $databasePlatform;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+        $this->databasePlatform = $connection->getDatabasePlatform();
```

Comment:
Needs to be fetched lazily because when container is compiled, db might not be available yet.

Reply:
Not sure about this: if we still aim to use `final` scope for metrics, they cannot be marked as `lazy`.

Reply:
> Not sure about this: if we still aim to use `final` scope for metrics, they cannot be marked as `lazy`.

@konradoboza  What I've meant is that `getDatabasePlatform` cannot be called in the constructor. It's to heavy and when db is not configured yet (composer create-project) it's likely to crash (depends on how the container is being built atm)

Fixed by common base, since you needed to pass `ezpublish.spi.persistence.connection` to make it repository-aware.
ref.: `\EzSystems\EzSupportTools\Storage\Metrics\RepositoryConnectionAwareMetrics`